src/Response/Elasticsearch.php

```diff
@@ -224,4 +227,63 @@ public function offsetUnset($offset): void
     {
         throw new ArrayAccessException('The array is reading only');
     }
+
+    /**
+     * Map the response body to an object of a specific class
+     * by default the class is the PHP standard one (stdClass)
+     * 
+     * This mapping works only for ES|QL results (with columns and values)
+     * @see https://www.elastic.co/guide/en/elasticsearch/reference/current/esql.html
+     * 
+     * @return object[] 
+     */
+    public function mapTo(string $class = stdClass::class): array
+    {
+        $response = $this->asArray();
+        if (!isset($response['columns']) || !isset($response['values'])) {
+            throw new UnknownContentTypeException(sprintf(
+                "The response is not a valid ES|QL result. I cannot mapTo(\"%s\")",
+                $class
+            )); 
+        }
+        $iterator = [];
+        $ncol = count($response['columns']);
+        foreach ($response['values'] as $value) {
+            $obj = new $class;
+            for ($i=0; $i < $ncol; $i++) {
+                $field = Utility::formatVariableName($response['columns'][$i]['name']);
+                if ($class !== stdClass::class && !property_exists($obj, $field)) {
+                    continue;
+                }
+                switch($response['columns'][$i]['type']) {
```

Comment:
Is there a file in Elasticsearch where we can see all the classes? It would be nice to add it in a comment if we have it, in case more types are added in the future.

Reply:
Yes, the list of data types managed by ES|QL is reported here: https://www.elastic.co/guide/en/elasticsearch/reference/current/esql-limitations.html#esql-supported-types